base/iobuffer.jl

```diff
@@ -260,22 +260,27 @@ bytesavailable(io::GenericIOBuffer) = io.size - io.ptr + 1
 position(io::GenericIOBuffer) = io.ptr - io.offset - 1
 
 function skip(io::GenericIOBuffer, n::Integer)
-    seekto = io.ptr + n
```

Comment:
I think it'd be better to replace this with `Base.Checked.add_with_overflow` and check the returned flag, rather than widen by default. That makes the intent of the code clearer. Then, if an overflow occurred, we can just return `io.size+1` (assuming it too doesn't overflow) and don't have to do any more wider arithmetic. If there is no overflow, the existing `min` call is still valid.